models/User.php

```diff
@@ -354,6 +357,44 @@ public function isBanned()
         return $throttle ? $throttle->is_banned : false;
     }
 
+    //
+    // IP Recording and Throttle
+    //
+
+    /**
+     * Records the last_ip_address to reflect the last known IP for this user.
+     * @return void
+     */
+    public function touchIpAddress($ipAddress)
+    {
+        $this
+            ->newQuery()
+            ->where('id', $this->id)
+            ->update(['last_ip_address' => $ipAddress])
+        ;
+    }
+
+    /**
+     * Returns true if IP address is throttled and cannot register
+     * again. Maximum 3 registrations every 60 minutes.
+     * @return bool
```

Comment:
@daftspunk As above:
```suggestion
     * @param string|null $ipAddress
     * @return bool
```